pkg/scanner/gitleaks_test.go

```diff
@@ -0,0 +1,59 @@
+package scanner
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/leaktk/scanner/pkg/config"
+	"github.com/leaktk/scanner/pkg/logger"
+	"github.com/leaktk/scanner/pkg/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+const mockGitleaksTestConfig = `
+[[rules]]
+id = "test-rule"
+description = "A rule for finding some value in the fake-leaks repo"
+regex = '''(?i)(fake)'''
+secretGroup = 1
+`
+
+func TestGitleaksScan(t *testing.T) {
+	err := logger.SetLoggerLevel("CRITICAL")
+	assert.NoError(t, err)
+
+	tempDir := t.TempDir()
+	configPath := filepath.Join(tempDir, "gitleaks.toml")
+	err = os.WriteFile(configPath, []byte(mockGitleaksTestConfig), 0644)
+	assert.NoError(t, err)
+
+	// Configured patterns
+	patterns := NewPatterns(
+		&config.Patterns{
+			Gitleaks: config.Gitleaks{
+				ConfigPath: configPath,
+			},
```

Comment:
This is where the issue I shared with you last week stems from. 

I suspect you have a config.toml somewhere it is expecting it, I did not. So i did not have the ExpiredAfter (or any others settings configured). Adding the following resolved it.
 
```suggestion
			},
			ExpiredAfter: 60 * 60 * 24 * 7,
```

Reply:
@thewizzy I went ahead and used `config.DefaultConfig()` here and in some other places. That should set that as well as some other relevant values. Does that also resolve the issue for you?

Reply:
Default config works well.